### ISSUE TITLE
[Bug] Documentation in botorch.utils.transforms.normalize is confusing/wrong

### ISSUE THREAD

**Issue by mc-robinson** [pytorch/botorch#215]
# 🐛 Bug

In the documentation for `botorch.utils.transforms.normalize`, it is written that the function is used to:

"Min-max normalize X to [0, 1] using the provided bounds."

However, this is slightly confusing, since the function will only transform X to [0,1] if the bounds provided represent the minimum and maximum of X. This ambiguous documentation string also features in the `unnormalize` function. 

Furthermore, the example given for the `normalize` function actually uses the `unnormalize` function as shown below:

```
Example:
        >>> X = torch.rand(4, 3)
        >>> bounds = torch.stack([torch.zeros(3), 0.5 * torch.ones(3)])
        >>> X_normalized = unnormalize(X, bounds)
```

I am happy to submit a pull request, but I think the point of this function should first be made clear. If the point is to provide min-max scaling with user-provided bounds -- then the one line at the beginning of the doc-string is all that needs changing. However, if the function is to provide min-max scaling to [0,1] given the data provided in X, `bounds` should either be automatically calculated, or the example should be something like:

```
X_bounds = torch.stack([X_train.min()*torch.ones(X_train.shape[1]),
                                          X_train.max()*torch.ones(X_train.shape[1])])     
X_train_norm = normalize(X_train, X_bounds)
```
or 
```
X_bounds = torch.stack([X_train.min(dim=1), X_train.max(dim=1)])     
X_train_norm = normalize(X_train, X_bounds)
```

Hope this is helpful!

**Comment by Balandat** [comment]
Thanks for spotting the issue in the example code. 

I guess you're also right that "normalizing to" is technically incorrect. We can change this to "normalizing w.r.t. the provided bounds". This is literally a line of code, so I guess I can just stick the quantity it returns into the docstring explicitly. 

I'll put up a PR for this.

**Comment by Balandat** [comment]
Fixed by #217
